opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/BaseInteractor.java

```diff
@@ -243,6 +244,9 @@ private void saveLocationInterventionForm(JSONObject jsonForm) {
                 jsonForm.put(ENTITY_ID, UUID.randomUUID().toString());
             }else if (CDD_SUPERVISOR_DAILY_SUMMARY.equals(encounterType)){
                 interventionType = CDD_SUPERVISION;
+            } else if(encounterType.equals(EventType.TABLET_ACCOUNTABILITY_EVENT)){
+                String location = JsonFormUtils.getFieldValue(jsonForm.toString(),JsonForm.LOCATION);
```

Comment:
Let's move this logic to inside the runnable on line 257 so that the call to the db can run in the background